pkg/controller/v1alpha1/trainedmodel/controller.go

```diff
@@ -251,6 +252,52 @@ func (r *TrainedModelReconciler) updateConditions(req ctrl.Request, tm *v1alpha1
 		}
 	}
 
+	// Get trained models with same inference service
+	var trainedModels v1alpha1api.TrainedModelList
+	if err := r.List(context.TODO(), &trainedModels, client.MatchingLabels{constants.ParentInferenceServiceLabel: isvc.Name, constants.TrainedModelMemoryIncluded: isvc.Name}); err != nil {
```

Comment:
Need to include the namespace in the ListOption, otherwise it may find the trained model in other namespaces with same inference service name.